File: avancado-intermediario/Python/Q5/objeto.py
from abc import ABC, abstractmethod

class Objeto(ABC):

    def __init__(self, x, y):
        self.__x = x
        self.__y = y

    def getX(self):
        return self.__x

    def setX(self, x):
        self.__x = x

    def getY(self):
        return self.__y

    def setY(self, y):
        self.__y = y

    @abstractmethod
    def __str__(self):
        return (f"Posição X = {self.getX()} \n"
                f"Posição Y = {self.getY()} \n")

File: avancado-intermediario/Java/Q5/Objeto.java
public abstract class Objeto {

    private double x;
    private double y;

    public Objeto() {
        this(0.0, 0.0);
    }

    public Objeto(double x, double y) {
        this.x = x;
        this.y = y;
    }

    public double getX() {
        return x;
    }

    public void setX(double x) {
        this.x = x;
    }

    public double getY() {
        return y;
    }

    public void setY(double y) {
        this.y = y;
    }

    @Override
    public String toString() {
        return "X = " + x +
               "\nY = " + y + "\n";
    }
}
